src/highlight.c

```diff
@@ -220,8 +220,8 @@ static char *(highlight_init_light[]) = {
     CENT("SignColumn term=standout ctermbg=Grey ctermfg=DarkBlue",
 	 "SignColumn term=standout ctermbg=Grey ctermfg=DarkBlue guibg=Grey guifg=DarkBlue"),
 #endif
-    CENT("Visual term=reverse",
-	 "Visual term=reverse guibg=LightGrey"),
+    CENT("Visual ctermbg=DarkGrey ctermfg=White",
+	 "Visual ctermbg=DarkGrey ctermfg=White guibg=#6b6b6b guifg=LightGrey"),
```

Comment:
If we are not to leave GUI visual as it was before,
I would have made foreground white here then:
![image](https://github.com/vim/vim/assets/234774/3f161e6a-fdf1-4e5c-87c1-54f2450c8b69)

Reply:
To reiterate on GUI visual hl, I would leave light bg the same as before (`guifg=NONE guibg=LightGrey`):

![image](https://github.com/vim/vim/assets/234774/0d3cc260-bd5e-499f-ad7b-bde0da2a6bb1)


And would made dark bg a bit more visible(`guifg=NONE guibg=#444444`):

![image](https://github.com/vim/vim/assets/234774/3d5937b5-cfab-4ad3-9c19-2f7e98c27a6d)


The main problem with default Visual is in terminal worlds where it is mostly unreadable.